src/main/java/com/blamejared/compat/tcomplement/highoven/MixRecipeHelper.java

```diff
@@ -0,0 +1,247 @@
+					+ LogHelper.getStackDescription(reducer) + " of type " + reducer.getClass().getSimpleName());
+		}
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper addPurifier(IIngredient purifier, int consumeChance) {
+		if (purifier instanceof IItemStack || purifier instanceof IOreDictEntry) {
+			this.purifiers.put(purifier, consumeChance);
+		} else {
+			CraftTweakerAPI.logWarning("addPurifier only supports IItemStack or IOreDictEntry, ignored purifier "
+					+ LogHelper.getStackDescription(purifier) + " of type " + purifier.getClass().getSimpleName());
+		}
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removeOxidizer(IIngredient oxidizer) {
+		this.oxidizers.remove(oxidizer);
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removeOReducer(IIngredient reducer) {
+		this.oxidizers.remove(reducer);
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removePurifier(IIngredient purifier) {
+		this.oxidizers.remove(purifier);
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removeAllOxidizer() {
+		this.oxidizers = new LinkedHashMap<IIngredient, Integer>();
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removeAllReducer() {
+		this.reducers = new LinkedHashMap<IIngredient, Integer>();
+		return this;
+	}
+
+	@ZenMethod
+	public MixRecipeHelper removeAllPurifier() {
+		this.purifiers = new LinkedHashMap<IIngredient, Integer>();
+		return this;
+	}
+
+	@ZenMethod
+	public void register() {
+		ModTweaker.LATE_ADDITIONS.add(
+				new AddMixRecipe(this.output, this.input, this.temp + 300, this.oxidizers, this.reducers, this.purifiers));
+	}
+
+	private static class AddMixRecipe extends BaseAction {
+		private FluidStack input, output;
+		private int temp;
+		private Map<IIngredient, Integer> oxydizers, reducers, purifiers;
+
+		public AddMixRecipe(ILiquidStack output, ILiquidStack input, int temp, Map<IIngredient, Integer> oxydizers,
+				Map<IIngredient, Integer> reducers, Map<IIngredient, Integer> purifiers) {
+			super("HighOven.MixRecipe");
+			this.input = InputHelper.toFluid(input);
+			this.output = InputHelper.toFluid(output);
+			this.temp = temp;
+			this.oxydizers = new LinkedHashMap<>();
+			this.oxydizers.putAll(oxydizers);
+			this.reducers = new LinkedHashMap<>();
+			this.reducers.putAll(reducers);
+			this.purifiers = new LinkedHashMap<>();
+			this.purifiers.putAll(purifiers);
+		}
+
+		@Override
+		public void apply() {
+			IMixRecipe recipe = TCompRegistry.registerMix(
+					new MixRecipeTweaker(this.input, this.output, this.temp));
+			for (Map.Entry<IIngredient, Integer> entry : this.oxydizers.entrySet()) {
+				if (entry.getKey() instanceof IItemStack)
+					recipe.addOxidizer(InputHelper.toStack((IItemStack) entry.getKey()), entry.getValue());
+				if (entry.getKey() instanceof IOreDictEntry)
+					recipe.addOxidizer(((IOreDictEntry) entry.getKey()).getName(), entry.getValue());
+			}
+			for (Map.Entry<IIngredient, Integer> entry : this.reducers.entrySet()) {
```

Comment:
This comes down to preference, but if you wanted to you could have just one list of additives and use `MixAdditive` to distinguish. Then when iterating the list you can just use the generic `recipe.addAdditive(MixAdditive,RecipeMatch)` method.

Reply:
`AddMixRecipe` now does that (but not `MixRecipeHelper`)